internal/core/project.go

```diff
@@ -60,6 +60,11 @@ type Project struct {
 	// overrideLabels are the labels specified via the CLI to override
 	// all other conflicting keys.
 	overrideLabels map[string]string
+
+	// variables are set on the job directly so that the runner
+	// can access this set of values along with any server- or vcs-stored
+	// values for final evaluation
+	variables []*pb.Variable
```

Comment:
This comment seems off or I'm just misunderstanding this. It seems like the core needs the variables fully rendered so that it can parse the configuration? i.e. they should be fully merged/flattened/whatever at this point.

Reply:
Yes, comment fixed.